compiler/rustc_middle/src/ty/query.rs

```diff
@@ -67,6 +71,24 @@ use std::sync::Arc;
 pub(crate) use rustc_query_system::query::QueryJobId;
 use rustc_query_system::query::*;
 
+pub struct QuerySystem<'tcx> {
+    pub local_providers: Box<Providers>,
+    pub extern_providers: Box<ExternProviders>,
```

Comment:
Why are the providers moved here?

Reply:
It was convenient to access them using a `TyCtxt` somewhere.

Reply:
I do want to try moving the rest of the `rustc_query_impl::Queries` state to `rustc_middle` though, as that slightly reduces the size of the query hot path.

Reply:
I'm sorry, I didn't see where you use them. Could you point it?

This data structure has been put into `rustc_query_impl` in an effort to shrink `rustc_middle` and its compile time. I'd rather see stuff moved to `rustc_query_impl`.

Reply:
To conclude this discussion, could you keep the current setup of fields in `GlobalCtxt`? To be re-discussed in another PR that needs those fields moved.